test/functional/p2p_permissions.py

```diff
@@ -146,7 +146,7 @@ def check_tx_relay(self):
             [tx],
             self.nodes[1],
             success=False,
-            reject_reason='Not relaying non-mempool transaction {} from forcerelay peer=0'.format(txid),
+            reject_reason='{} from peer=0 was not accepted: txn-mempool-conflict'.format(txid)
```

Comment:
I think to fully test that sending an invalid transaction from a forcerelay peer doesn't result in the tx being relayed, you'd need to add:

```diff
+        p2p_rebroadcast_wallet.send_txs_and_test(
+            [tx],
+            self.nodes[1],
+            success=False,
+            reject_reason='Not relaying non-mempool transaction {} from forcerelay peer=0'.format(txid)
+        )
+
```

The first time the transaction is sent, it's rejected and not added to the mempool, but is added to recentRejects. The second time the transaction is sent, we AlreadyHave it and choose not to relay it because it's not in the mempool.

Reply:
thanks, fixed

Reply:
You should also add a comment here to explain why you're sending the transaction twice.